### ISSUE TITLE
Nested batches

### ISSUE THREAD

**Issue by cybey** [breamware/sidekiq-batch#5]
I noticed in the "Projects" that this is a wanted feature.

Any chance of this being done soon? If not, could you point me in the right direction on how you would see this being developed? What the right approach should be?

The problem I'm trying to solve is as follows:

WorkflowWorker will spawn 1 overall batch, which will spawn X batches (from database, so don't know how many).
Each batch will create a new batch of 1 worker, which when finished (callback), will start a new batch with X workers (from database, so don't know how many).

When everything's finished, the WorkflowWorker overall batch should call a callback, spawning a final worker.

So, basically. If [this](https://github.com/mperham/sidekiq/wiki/Really-Complex-Workflows-with-Batches) implementation succeeds, I would be able to do what I want.

**Comment by cybey** [comment]
Update: taking a crack at this. Really close...

**Comment by managr** [comment]
So @cybey I didn't think it over much before but maybe the simplest solution is to pass the ids of the parent batches to a child one and just add the job id to the parent batch (including child batch too). Also you'll have take care of notifying the parent batch to know when the jobs are ready.
Let me know if you need some help.

**Comment by cybey** [comment]
https://github.com/cybey/sidekiq-batch/commits/nested_batches

Code is still a bad rough. It seems to be working for completed and successful event. However, failures still don't work as they should.

**Comment by cybey** [comment]
Test have been rewritten and a few added. The issue I had with failed callbacks have been resolved.

Will go through it again tomorrow and create a PR for it if you don't mind.

**Comment by managr** [comment]
That's great! Sure, open a PR whenever you think it's ready. We can discuss it then in the PR.

**Comment by cybey** [comment]
Have opened a PR at #6

**Comment by managr** [comment]
Thanks @cybey will look into it.

**Comment by managr** [comment]
Closing as 0.1.3 was released. Appreciate @cybey
